### DISCUSSION TITLE
How to tint a sprite based on Texture.WHITE?

### DISCUSSION THREAD

**Discussion by dbuzinin** [pixijs/pixijs#9375] [Q&A] [answered]
Hi! I have a sprite created from a white texture. To color the sprite, I use sprite.tint = 
`const sprite = new Sprite(Texture.WHITE)`
`sprite.tint = COLOR_GREEN;`

I need to get an effect that is applied to the sprite if it was already colored. For example the sprite was tinted green, I tinted it light blue and the final color would be green with a blue tint.

![image](https://user-images.githubusercontent.com/75254600/232105472-aef7e6a8-d69b-4bad-a827-d80e8452de91.png)
![image](https://user-images.githubusercontent.com/75254600/232105515-9bea4684-e329-4560-9e35-96a37d3860d5.png)

**Comment by bigtimebuddy** [comment]
Something like this?
https://pixiplayground.com/#/edit/We_ZTkKdl6zVQPs92hPun

**Comment by dbuzinin** [comment]
I update your example.

I have a sprite with green color (sprite2) and for example on the event hover I want to have it in addition to the green color blue (sprite3). But if the sprite was created based on Texture.WHITE then the color is not mixed, and applied to it.

And now I'm looking for functions to get the same effect when the colors are mixed, but so far without success. Found on the internet examples of mixing colors, do not give the same shade as the sprite3.

https://pixiplayground.com/#/edit/pRX-grwOWJXb5qJIOEd2i

**Comment by dbuzinin** [comment, marked as answer]
I found a solution to my question.

```js
function blendColors(color1, color2) {
  const hexToRgb = (hex) => {
    const r = parseInt(hex.substring(1, 3), 16);
    const g = parseInt(hex.substring(3, 5), 16);
    const b = parseInt(hex.substring(5, 7), 16);
    return { r, g, b };
  };

  const rgbToHex = (rgb) => {
    const rHex = rgb.r.toString(16).padStart(2, '0');
    const gHex = rgb.g.toString(16).padStart(2, '0');
    const bHex = rgb.b.toString(16).padStart(2, '0');
    return '#' + rHex + gHex + bHex;
  };

  const rgb1 = hexToRgb(color1);
  const rgb2 = hexToRgb(color2);

  const r = Math.round((rgb1.r * rgb2.r) / 255);
  const g = Math.round((rgb1.g * rgb2.g) / 255);
  const b = Math.round((rgb1.b * rgb2.b) / 255);

  const resultHex = rgbToHex({ r, g, b });

  return resultHex;
}

blendColors('#7acc90', '#a6c4fa') // '#4f9d8d'

```
